Lib/re/_parser.py

```diff
@@ -1004,26 +1004,21 @@ def addgroup(index, pos):
             # group
             c = this[1]
             if c == "g":
-                name = ""
                 if not s.match("<"):
                     raise s.error("missing <")
                 name = s.getuntil(">", "group name")
-                if name.isidentifier():
+                if (name.isdecimal() and name.isascii() and
+                        (name[0] != "0" or name == "0")):
+                    index = int(name)
+                    if index >= MAXGROUPS:
+                        raise s.error("invalid group reference %d" % index,
+                                      len(name) + 1)
+                else:
+                    s.checkgroupname(name, 1)
                     try:
                         index = groupindex[name]
```

Comment:
Maybe this could also use `.get(name)` and if to be more consistent with the the similar error raised above (line 798).

Reply:
I left it to other issue. I want to change `IndexError` to `re.error`.